File: server/src/Models/DMToolkitModels/Instances/SpeciesInstance.cs
using DMToolkit.API.Models.DMToolkitModels.Definitions;

namespace DMToolkit.API.Models.DMToolkitModels.Instances;

public class SpeciesInstance : IDefinitionInstance<SpeciesDefinition>
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Size { get; set; } = string.Empty;
    public string LineageInstanceId { get; set; } = string.Empty;
    public LineageInstance? LineageInstance { get; set; } = null;
    public ICollection<FeatInstance> FeatInstances { get; set; } = new List<FeatInstance>();
    public string DefinitionId { get; set; } = string.Empty;
    public SpeciesDefinition? Definition { get; set; } = null;
}

File: web-client/src/lib/types/dm-tool-types/instances/speciesInstance.ts
import type { SpeciesDefinition } from "../definitions/speciesDefinition";
import type { FeatInstance, FeatInstanceDto } from "./featInstance";
import type { LineageInstance, LineageInstanceDto } from "./lineageInstance";

export interface SpeciesInstance {
    id: string;
    size: string;
    lineageInstance: LineageInstance | null;
    featInstances: FeatInstance[];
    definition: SpeciesDefinition;
}

export interface SpeciesInstanceDto {
    id: string;
    size: string;
    lineageInstance: LineageInstanceDto | null;
    featInstances: FeatInstanceDto[];
    definitionId: string;
}